azkaban-common/src/main/java/azkaban/utils/CaseInsensitiveConcurrentHashMap.java

```diff
@@ -0,0 +1,39 @@
+/*
+* Copyright 2018 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+package azkaban.utils;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CaseInsensitiveConcurrentHashMap<V> extends ConcurrentHashMap<String, V> {
+
+  @Override
+  public V put(final String key, final V value) {
+    return super.put(key.toLowerCase(), value);
+  }
+
+  public V get(final String key) {
```

Comment:
Missing @Override annotations?

Reply:
The get method in the parent class requires Object as the argument. Here in the child class, I want to specify String type as the argument because the case-insensitive key comparison (toLowerCase()) only applies to String.

Reply:
I see. Thanks for the explanation.

The concurrentHashMap is a complex class. I looked at the documentation. It doesn't seem to indicate that it is designed for inheritance. Are you sure it is safe to override these methods? Is it going to violate some invariance? 
It's one of the reasons I feel more comfortable with an existing library which has been tried in production. 

If you would like to keep this approach, I'd suggest you think about forwarding the calls to an internal private instance of the concurrentHashMap. This way you may have higher confidence that this code works by reducing the APIs your class exposes.

Reply:
+1 
- Favor composition over inheritance.
https://github.com/azkaban/azkaban/blob/master/CONTRIBUTING.md

Reply:
Thanks for the suggestions! I've updated the PR.